src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -47,13 +48,17 @@ public CategoryDto create(CreateCategoryRequestDto requestDto) {
 
     @Override
     public CategoryDto updateById(CreateCategoryRequestDto requestDto, Long id) {
+        categoryRepository.findById(id)
```

Comment:
Why do you need to fetch a whole entity if you don't use it? simply call `existsById` for this purpose